### DISCUSSION TITLE
NumericIDs migration fails in memory constrained environments

### DISCUSSION THREAD

**Discussion by RealOrangeOne** [plausible/analytics#3236] [Self-Hosted Support]
### Past Issues Searched

- [X] I have searched open and closed issues to make sure that the bug has not yet been reported

### Issue is a Bug Report

- [X] This is a bug report and not a feature request, nor asking for self-hosted support

### Using official Plausible Cloud hosting or self-hosting?

Self-hosting

### Describe the bug

Attempting to run the `NumericIDs` migration fails if there isn't enough RAM on the server, which prevents an upgrade to 2.0

The migration clearly loads a lot of data into memory. It happily seems to use about 1GB or more instantly, and the container crashes. 

There are no minimum requirements for Plausible, and I've had the application (and Clickhouse) running on a single 2GB VM for a few years now.

### Expected behavior

I can run `NumericIDs` migration and upgrade to Plausible 2.0.

**Comment by ruslandoga** [comment]
👋 @RealOrangeOne 

> There are no minimum requirements for Plausible

But there are recommendations: https://plausible.io/docs/self-hosting#requirements which mostly come from ClickHouse docs.

**Reply by RealOrangeOne** [reply]
Interesting, I have no idea how I missed this - thanks!

**Comment by ruslandoga** [comment]
But that shouldn't matter, I haven't had problems with the data migration on my 2GB instance. I wonder how much data you have, how many events, how many sites, etc. At which point in the migration process the OOM occurs?

**Reply by ruslandoga** [reply]
Maybe the container gets OOM-killed for a different reason, like using city-level geolocations? Those are loaded in memory.

**Reply by RealOrangeOne** [reply]
Running afresh from inside, I'm getting a new error:

```
$  docker compose exec plausible bin/plausible rpc Plausible.DataMigration.NumericIDs.run
--rpc-eval : RPC failed with reason :nodedown
```

**Reply by ruslandoga** [reply]
https://github.com/plausible/analytics/discussions/2896#discussioncomment-5986941 is probably relevant: `RELEASE_DISTRIBUTION=none` should not be set for rpc to work.

**Reply by RealOrangeOne** [reply]
Very relevant - unsetting that has fixed the issue and the migration runs clean. I'm not sure what was happening about it displaying no output previously.
